include/api/Metrics.h

```diff
@@ -289,10 +280,126 @@ class Metrics
 
   mutable std::mutex _mutex;
   LookupTable _lookups;
-  MetricBlobs _blobs;
+  BlobStorage _blobs;
   uint16_t _cur_blob = 0;
   uint16_t _cur_off  = 0;
 
+public:
+  // These are sort of factory classes, using the Metrics singleton for all storage etc.
+  class Gauge
+  {
+  public:
+    using self_type = Gauge;
+    using SpanType  = Metrics::SpanType;
+
+    class AtomicType : public Metrics::AtomicType
+    {
+    };
+
+    static Metrics::IdType
+    create(const std::string_view name)
+    {
+      auto &instance = Metrics::instance();
+
+      return instance._create(name);
+    }
+
+    static AtomicType *
+    createPtr(const std::string_view name)
+    {
+      auto &instance = Metrics::instance();
+
+      return reinterpret_cast<AtomicType *>(instance.lookup(instance._create(name)));
```

Comment:
This looks like the overload of `lookup()` that returns `IdType/uint32_t`.  How can that work, to cast it to a64-bit pointer?

Reply:
That's a different lookup no? I'd have to look again, but remember there are two lookups, one to that returns the IdType, and one that returns a pointer directly to the atomic types.

Reply:
OK I see but it shouldn't need the case then.